### DISCUSSION TITLE
ot.da.SinkhornTransport : does barycentric transform work for metric!='sqeuclidean'?

### DISCUSSION THREAD

**Discussion by samarth4149** [PythonOT/POT#752] [Q&A]
Hi,

I'm unsure whether the transform the way it is currently computed using the coupling matrix would work for a metric which is not the l_2 norm going by equations 13 and 14 [here](https://arxiv.org/pdf/1507.00504.pdf). Would this for instance, work with cosine distance metric? 
Cosine also seems to pose the following question : Would the final transformed source examples then be correct up to a multiplicative factor?

Much appreciated!

**Comment by rflamary** [comment]
This is a very good question!

No it does not! While the euclidean barycentric mapping makes sens it is indeed not the proper barycentric mapping for any loss. We should definitely add this to the documentation and begin implementing for other losses (which is much more complex since it will require to solve optimization problems on several of them)
